### PR TITLE
Add side conditions for inequalities over BITVECTOR_UREM for CBQI BV.

### PR DESCRIPTION
**Pull request by aniemetz**
We now can handle all cases of (in|dis)equality over UREM. Previously, we could not handle equality for index=0 and had to rewrite `x % s = t` to `x - x / s * s`. Since we can now handle this case, we do not apply this rewriting anymore.

### CODE REVIEW

File: src/theory/quantifiers/bv_inverter.cpp [hunk 1]
```diff
@@ -376,54 +375,168 @@ static Node getScBvUrem(bool pol,
...
+        Node o2 = nm->mkNode(BITVECTOR_NEG, t);
+        Node o = nm->mkNode(BITVECTOR_OR, o1, o2);
+        scl = nm->mkNode(BITVECTOR_SLT, nm->mkNode(BITVECTOR_NOT, t), o);
+      }
+      else
+      {
+        /* x % s >= t
+         * with side condition (synthesized):
+         * (not (and (not (bvslt t s)) (bvslt #x0 s)))  */
+        Node slt1 = nm->mkNode(BITVECTOR_SLT, t, s).notNode();
```
Comment (mpreiner) [inline src/theory/quantifiers/bv_inverter.cpp]: Use BITVECTOR_SGE.

File: src/theory/quantifiers/bv_inverter.cpp [hunk 2]
```diff
@@ -376,54 +375,168 @@ static Node getScBvUrem(bool pol,
...
+        Node o1 = nm->mkNode(BITVECTOR_NEG, s);
+        Node o2 = nm->mkNode(BITVECTOR_NEG, t);
+        Node o = nm->mkNode(BITVECTOR_OR, o1, o2);
+        scl = nm->mkNode(BITVECTOR_SLT, nm->mkNode(BITVECTOR_NOT, t), o);
+      }
+      else
+      {
+        /* x % s >= t
+         * with side condition (synthesized):
+         * (not (and (not (bvslt t s)) (bvslt #x0 s)))  */
```
Comment (mpreiner) [inline src/theory/quantifiers/bv_inverter.cpp]: Would it be better to use `(or (bvslt t s) (bvsge #x0 s))` instead?

Comment (aniemetz) [reply src/theory/quantifiers/bv_inverter.cpp]: Indeed. Done.

File: src/theory/quantifiers/bv_inverter.cpp (hunk 1 above)
Comment (aniemetz) [reply src/theory/quantifiers/bv_inverter.cpp]: Obsolete (see comment above).

**Pull request merged**